pkgs/by-name/li/libwacom/package.nix

```diff
@@ -0,0 +1,94 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  meson,
+  ninja,
+  glib,
+  pkg-config,
+  udev,
+  libevdev,
+  libgudev,
+  python3,
+  valgrind,
+}:
+
+let
+  testsDisabled = "-Dtests=disabled";
+  testsEnabled = "-Dtests=enabled";
+  replaceDisabledWithEnabled =
+    flag:
+    lib.replaceStrings [
+      testsDisabled
+      testsEnabled
+    ] flag;
+in
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "libwacom";
+  version = "2.12.2";
+
+  outputs = [
+    "out"
+    "dev"
+  ];
+
+  src = fetchFromGitHub {
+    owner = "linuxwacom";
+    repo = "libwacom";
+    rev = "libwacom-${finalAttrs.version}";
+    hash = "sha256-dxnXh+O/8q8ShsPbpqvaBPNQR6lJBphBolYTmcJEF/0=";
+  };
+
+  postPatch = ''
+    patchShebangs test/check-files-in-git.sh
+  '';
+
+  nativeBuildInputs = [
+    pkg-config
+    meson
+    ninja
+    python3
+  ];
+
+  buildInputs = [
+    glib
+    udev
+    libevdev
+    libgudev
+  ];
+
+  # Tests are in the `tests` pass-through derivation.
+  # See https://github.com/NixOS/nixpkgs/issues/328140
+  doCheck = false;
+  mesonFlags = [
+    testsDisabled
+    "--sysconfdir=/etc"
+  ];
+
+  passthru = {
+    tests = finalAttrs.finalPackage.overrideAttrs (prevAttrs: {
```

Comment:
This would be much better achieved using an override with an explicit API rather than directly fumbling with derivation internals using overrideAttrs.

The previous iteration worked like that AFAICT but it used overrideAttrs + finalAttrs to build such an API upon an mkDerivation implementation detail (doCheck) which I consider a bad pattern.

Instead I'd put a `withChecks ? false` in the args up top and make the drv respond to that boolean. Then override in the passthru using the abstract .override API. Much better encapsulation that way.

Reply:
I considered doing it that way and went with the most simple thing, which was futzing with the inputs. I'd be happy to switch to `withChecks` or equivalent name.

As a matter of bikeshedding, `withChecks` is used once in [`glib`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/glib/default.nix#L305). It uses the standard `doCheck` flag to turn on and off tests. I could do that.

No package uses `enableChecks`.

[wayland](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/wayland/default.nix#L11) and [systemd](https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/systemd/default.nix#L162) both use `withTests`.

Reply:
> I considered doing it that way and went with the most simple thing, which was futzing with the inputs.

I do not see how defining functions for modifying flags and applying them optionally is simpler than simply defining the flags in terms of the optional value while requiring invasion on implementation details.

I'll just implement what I mean and PR your branch, I think you'll like it better to.

> As a matter of bikeshedding, `withChecks` is used once in [`glib`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/glib/default.nix#L305). It uses the standard `doCheck` flag to turn on and off tests. I could do that.
> 
> No package uses `enableChecks`.
> 
> [wayland](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/wayland/default.nix#L11) and [systemd](https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/systemd/default.nix#L162) both use `withTests`.
> 
> [libdicom](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/libdicom/default.nix#L35) uses just `doCheck`. I think I like that the most.

I don't care about the exact name, it just needs to be obvious that it decides whether tests are run or not.

Reply:
> I do not see how defining functions for modifying flags and applying them optionally while requiring invasion on implementation details is simpler than defining the flags in terms of the optional value.

Oh, I'm agreeing with you. Simple in this case should be read as "able to be typed out without thinking too hard about it because I know I'll get feedback" not "actually simple". 😬

Reply:
https://github.com/philiptaron/nixpkgs/pull/1

Reply:
Thanks, merged.